### PR TITLE
Modified response code for ConstraintViolation

### PR DESCRIPTION
**Pull request by akifkhan01**
In case of constraint violation, changed response code to 500
(INTERNAL_SERVER_ERROR) if it is not for duplicate State Machine Id.
For duplciate state machine id, it sends 409 (CONFLICT) as response.

All tests passing.

### CODE REVIEW

File: runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java [hunk 1]
```diff
@@ -160,11 +159,22 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
                     append(stateMachine.getName()).
                     append(".started").toString());
         } catch (ConstraintViolationException ex) {
-            //in case of Duplicate correlation key, return http code 409 conflict
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(ex.getCause() != null ? ex.getCause().getMessage() : null).build();
+            //In case of Duplicate correlation key, return http code 409 conflict
+            stateMachine1 = stateMachinesDAO.findById(stateMachineInstanceId);
```
Comment (amitkumar94) [inline runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: Check when ex.getMessage() contains DuplicatePrimary Key else throw 5xx, don't make a find call

Comment (akifkhan01) [reply runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: Done.

File: runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java [hunk 2]
```diff
@@ -139,7 +138,7 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
         if (stateMachineDefinition == null)
             throw new IllegalRepresentationException("State machine definition is empty");
 
-        StateMachine stateMachine = null;
+        StateMachine stateMachine = null, stateMachine1 = null;
```
Comment (amitkumar94) [inline runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: stateMachine1 is no longer needed

Comment (akifkhan01) [reply runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: Removed.

File: runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java [hunk 3]
```diff
@@ -160,11 +159,23 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
                     append(stateMachine.getName()).
                     append(".started").toString());
         } catch (ConstraintViolationException ex) {
-            //in case of Duplicate correlation key, return http code 409 conflict
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(ex.getCause() != null ? ex.getCause().getMessage() : null).build();
+            //In case of Duplicate correlation key, return http code 409 conflict
+            if(ex.getCause() != null) {
+                if (ex.getCause().getMessage().contains("Duplicate entry")) {
```
Comment (ykameshrao) [inline runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: ex.getClause().getMessage().toLowerCase().contains("duplicate entry") 

make it case agnostic.

Comment (ykameshrao) [reply runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: can you instead do this in the underlying DAO layer and throw relevant exception from their? DATA layer details are leaking to Resource layer which is exactly what DAO was supposed to abstract out.

Comment (akifkhan01) [reply runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: > ex.getClause().getMessage().toLowerCase().contains("duplicate entry")
> 
> make it case agnostic.

Done, thanks for pointing out.

File: runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java [hunk 4]
```diff
@@ -160,11 +159,23 @@ public Response createStateMachine(StateMachineDefinition stateMachineDefinition
                     append(stateMachine.getName()).
                     append(".started").toString());
         } catch (ConstraintViolationException ex) {
-            //in case of Duplicate correlation key, return http code 409 conflict
-            return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(ex.getCause() != null ? ex.getCause().getMessage() : null).build();
+            //In case of Duplicate correlation key, return http code 409 conflict
+            if(ex.getCause() != null) {
+                if (ex.getCause().getMessage().toLowerCase().contains("duplicate entry")) {
+                    return Response.status(Response.Status.CONFLICT.getStatusCode()).entity(
+                            ex.getCause() != null ? ex.getCause().getMessage() : null).build();
```
Comment (ykameshrao) [inline runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: isn't this null check redundant? you already checked at line 163.

Comment (akifkhan01) [reply runtime/src/main/java/com/flipkart/flux/resource/StateMachineResource.java]: Removed redundant check.

**Pull request merged**